src/librustc_trans/back/write.rs

```diff
@@ -667,7 +667,8 @@ pub fn run_passes(sess: &Session,
 
     // Sanity check
     assert!(trans.modules.len() == sess.opts.cg.codegen_units ||
-            sess.opts.debugging_opts.incremental.is_some());
+            sess.opts.debugging_opts.incremental.is_some() ||
+            output_types.contains_key(&OutputType::Metadata));
```

Comment:
This is just replicating the condition from here: 
https://github.com/rust-lang/rust/blob/47965f51e6d91ae33bf2b0da0f5510f03766feac/src/librustc_trans/base.rs#L1186

It's not an important assertion anyway.